runtime/assembly/modules/storage.ts

```diff
@@ -36,4 +38,48 @@ export namespace Storage {
         const value64: u64 = Serialiser.serialiseResultwOutRetain(value);
         ext_storage_set_version_1(key64, value64);
     }
+    /**
+     * Gets the given key from storage, placing the value into a buffer and returning the number of bytes that the
+     * entry in storage has beyond the offset. 
+     * ArrayBuffer has fixed length of 20.
+     * Returns Option instance with populated ArrayBuffer concatenated to the result from storage_read
+     * @param key key of the data to get
+     * @param offset offset beyond which the value is copied
+     */
+    export function read(key: u8[], offset: i32): Option<u8[]> {
```

Comment:
I think that it would be beneficial to have an `assert` that checks that the `offset` that is passed is `<=` to the length of the `arraybuffer` passed.

If the buffer is 10 and the offset is 11, we won't read anything and it will be pointless right?

Reply:
Not exactly. The length of the value could be let's say 30, and after the 11, there are 19 elements and the first 10 would be written to the `buffer`. 
Also, the docs say: 
`If value_out length is smaller than the returned length, only value_out length bytes are copied into value_out.`
Better assertion imo, would be to check if the `offset` is not bigger than the length of the value. But I think it would be boilerplate to add that.

Reply:
well if the offset is not a constant but a dynamically calculated value along with the array buffer (coming from code that we are building on top) it may be the case that by letting people pass offset bigger than the array buffer we are hiding bugs without triggering anything. That is why I think it's a good idea to have that assert

Reply:
But offset can be bigger than the length of the array buffer, no? Again, the last example, offset is 30, array buffer length is 10, but the length of the value is 40. So the elements from 30th to 39th index will be written to the buffer.

Reply:
oh yeah sorry. my bad